systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java

```diff
@@ -330,4 +333,72 @@ void testSwitchingStrimziPodSetFeatureGateOnAndOff(ExtensionContext extensionCon
+        final LabelSelector kafkaSelector = KafkaResource.getLabelSelector(clusterName, KafkaResources.kafkaStatefulSetName(clusterName));
+
+        int messageCount = 180;
+        List<EnvVar> testEnvVars = new ArrayList<>();
+        int kafkaReplicas = 3;
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+UseStrimziPodSets,+UseKRaft", null));
+
+        clusterOperator.unInstall();
+        clusterOperator = new SetupClusterOperator.SetupClusterOperatorBuilder()
+                .withExtensionContext(BeforeAllOnce.getSharedExtensionContext())
+                .withNamespace(INFRA_NAMESPACE)
+                .withWatchingNamespaces(Constants.WATCH_ALL_NAMESPACES)
+                .withExtraEnvVars(testEnvVars)
+                .createInstallation()
+                .runInstallation();
+
+        Kafka kafka = KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build();
+        kafka.getSpec().setEntityOperator(null); // The builder cannot disable the EO. It has to be done this way.
+        resourceManager.createResource(extensionContext, kafka);
+
+        LOGGER.info("Try to send some messages to Kafka over next few minutes.");
+
+        KafkaClients kafkaBasicClientJob = new KafkaClientsBuilder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(messageCount)
+                .withDelayMs(500)
+                .withNamespaceName(INFRA_NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi());
+        JobUtils.waitForJobRunning(consumerName, INFRA_NAMESPACE);
+
+        // Check that there is no ZooKeeper
+        Map<String, String> zkPods = PodUtils.podSnapshot(INFRA_NAMESPACE, zkSelector);
+        assertThat("No ZooKeeper pods should exist", zkPods.size(), is(0));
+
+        // Roll Kafka
+        LOGGER.info("Force Rolling Update of Kafka via read-only configuration change.");
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(INFRA_NAMESPACE, kafkaSelector);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, k -> k.getSpec().getKafka().getConfig().put("log.retention.hours", 72), INFRA_NAMESPACE);
+
+        LOGGER.info("Wait for next reconciliation to happen.");
+        RollingUpdateUtils.waitTillComponentHasRolled(INFRA_NAMESPACE, kafkaSelector, kafkaReplicas, kafkaPods);
+
+        LOGGER.info("Waiting for clients to finish sending/receiving messages.");
+        ClientUtils.waitForClientSuccess(producerName, INFRA_NAMESPACE, MESSAGE_COUNT);
+        ClientUtils.waitForClientSuccess(consumerName, INFRA_NAMESPACE, MESSAGE_COUNT);
```

Comment:
```suggestion
        ClientUtils.waitForClientsSuccess(producerName, consumerName, INFRA_NAMESPACE, MESSAGE_COUNT);
```

Reply:
or with `testStorage.getProducerName()` and `testStorage.getConsumerName()` when you use the `TestStorage`